lib/game_teams_manager.rb

```diff
@@ -112,33 +109,24 @@ def rival(team_id)
     rival_hash_ratio.key(rival_hash_ratio.values.max)
   end
 
-  def total_goals_by_team
-    goals_by_team_id = Hash.new(0)
-    game_teams.each do |game|
-      goals_by_team_id[game.team_id] += game.goals
-    end
-    goals_by_team_id
-  end
-
-  def total_games_by_team
-    games_by_team_id = Hash.new(0)
-    game_teams.each do |game|
-      games_by_team_id[game.team_id] += 1
+  def create_goals_hash
+    goals_hash = Hash.new { |coach, team| coach[team] = [0,0] }
+    @game_teams.each do |game_team|
+      goals_hash[game_team.team_id][1] += 1
+      goals_hash[game_team.team_id][0] += game_team.goals
     end
-    games_by_team_id
+    goals_hash
   end
 
   def best_offense
-    averages = total_goals_by_team.merge(total_games_by_team) do |team_id, goals, games|
-      get_percentage(goals, games)
-    end
-    averages.max_by {|team_id, average| average}.first
+    goal_hash = create_goals_hash
+    goal_hash_ratio = create_ratio_hash(goal_hash)
+    goal_hash_ratio.max_by {|team_id, average| average}.first
   end
 
   def worst_offense
-    averages = total_goals_by_team.merge(total_games_by_team) do |team_id, goals, games|
-      get_percentage(goals, games)
-    end
-    averages.min_by {|team_id, average| average}.first
+    goal_hash = create_goals_hash
+    goal_hash_ratio = create_ratio_hash(goal_hash)
```

Comment:
maybe put  goal_hash_ratio = create_ratio_hash(create_goals_hash)
to avoid creating that var?